flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/catalog/DatabaseCalciteSchema.java

```diff
@@ -86,6 +86,7 @@ private FlinkStatistic getStatistic(
         final ResolvedCatalogBaseTable<?> resolvedBaseTable = lookupResult.getResolvedTable();
         switch (resolvedBaseTable.getTableKind()) {
             case TABLE:
+            case MANAGED:
```

Comment:
I thought about my previous comment around a special `TableKind` again. I fear we are mixing metadata encoded in table kind and metadata encoded in options.